### PR TITLE
Migrate app-package to type module

### PR DESCRIPTION
**Pull request by domachine**
This PR migrates Secvisogram's app package to type `"module"` which aligns it with modern package standards. This is needed to integrate the new csaf-validator-lib.

### CODE REVIEW

File: app/cypress/fixtures/documentTests.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import minimalDoc from '../shared/minimalCSAFBaseDoc'
-import minimalInformationalAdvisoryDoc from '../shared/minimalInformationalAdvisoryDoc'
-import minimalSecurityAdvisoryDoc from '../shared/minimalSecurityAdvisoryDoc'
-import minimalSecurityIncidentResponseDoc from '../shared/minimalSecurityIncidentResponseDoc'
-import minimalVexDoc from '../shared/minimalVexDoc'
+import minimalDoc from './documentTests/minimalGenericCSAFDoc'
```
Comment (tschmidtb51) [inline app/cypress/fixtures/documentTests.js:1]: I guess that must be `minimalCSAFBaseDoc`. See #234 

´´´suggestion
import minimalDoc from './documentTests/minimalCSAFBaseDoc'

File: app/cypress/fixtures/documentTests.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+import minimalInformationalAdvisoryDoc from './documentTests/minimalInformationalAdvisoryDoc'
+import minimalSecurityAdvisoryDoc from './documentTests/minimalSecurityAdvisoryDoc'
+import minimalSecurityIncidentResponseDoc from './documentTests/minimalSecurityIncidentResponseDoc'
+import minimalVexDoc from './documentTests/minimalVexDoc'
```
Comment (tschmidtb51) [inline app/cypress/fixtures/documentTests.js:5]: Why don't we need the `.js` here? We needed it in all other files...

File: app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import minimalDoc from './shared/minimalDoc'
+
+export default {
+  ...minimalDoc,
+  document: {
+    ...minimalDoc.document,
+    category: 'csaf_base',
+  },
+}
```
Comment (tschmidtb51) [inline app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js:9]: That file should be named `app/cypress/fixtures/documentTests/minimalCSAFBaseDoc.js`

**Review by tschmidtb51 — CHANGES_REQUESTED** [review]
Some minor questions / CR from my side to align with #240 .

File: app/cypress/fixtures/documentTests.js (hunk 2 above)
Comment (domachine) [reply app/cypress/fixtures/documentTests.js:5]: Good catch! It seems that the webpack bundled with cypress doesn't care about the `"type": "module"` in the package.json. But I'll keep it consistent.

File: app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js (hunk 3 above)
Comment (domachine) [reply app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js:9]: Yes, thank you! This conflicted when rebasing on the main.

File: app/cypress/fixtures/documentTests.js (hunk 1 above)
Comment (tschmidtb51) [reply app/cypress/fixtures/documentTests.js]: Done in [3e5687f](https://github.com/secvisogram/secvisogram/pull/241/commits/3e5687f006764a7938f8d747c8d860d47e732920).

File: app/cypress/fixtures/documentTests.js (hunk 2 above)
Comment (tschmidtb51) [reply app/cypress/fixtures/documentTests.js]: Done in [3e5687f](https://github.com/secvisogram/secvisogram/pull/241/commits/3e5687f006764a7938f8d747c8d860d47e732920).

File: app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js (hunk 3 above)
Comment (tschmidtb51) [reply app/cypress/fixtures/documentTests/minimalGenericCSAFDoc.js]: Done in [3e5687f](https://github.com/secvisogram/secvisogram/pull/241/commits/3e5687f006764a7938f8d747c8d860d47e732920).

**Review by tschmidtb51 — APPROVED** [review]
LGTM.

@domachine: Just out of curiosity: Why do we rename some files from `.js` to `.cjs`?

**Comment by domachine** [comment]
> LGTM.
> 
> @domachine: Just out of curiosity: Why do we rename some files from `.js` to `.cjs`?

`.cjs` is the common js compatibility extension of nodejs: https://nodejs.org/api/esm.html#interoperability-with-commonjs . These are configuration files from tools, that not yet support the ecma module system.

**Review by cloeser — APPROVED** [review]
LGTM

**Comment by tschmidtb51** [comment]
@domachine Just fixed the permission issue. This seems to be ready to merge :-)

**Pull request merged**